web/unit_tests/cases/engine.js

```diff
@@ -40,6 +40,25 @@ describe('Engine - Browser Interactions', function() {
     });
   });
 
+  describe('RegisterStub', function() {
+    it.only('RegisterStub 2x on same keyboard', function(done) {
+      this.timeout(kmwconfig.timeouts.scriptLoad);
+      var keyman = com.keyman.singleton;
+
+      var stub = {
+        'KI': 'Keyboard_lao_2008_basic',
+        'KN': 'Lao 2008 Basic',
+        'KLC': 'lo',
+        'KL': 'Lao',
+        'KF': 'resources/keyboards/lao_2008_basic.js'
+      };
+      assert.isNull(com.keyman.text.KeyboardInterface.prototype.registerStub(stub), "Keyboard stub was not registered!");
+      assert.equal(com.keyman.text.KeyboardInterface.prototype.registerStub(stub), 1, "Registering existing keyboard should return 1!");
```

Comment:
@jahorton I tried to create a unit test that calls registerstub() twice.

setActiveKeyboard mysteriously fails with what looks like the beginning of lao_2008_basic keyboard
```
if(typeof keyman === 'undefined') {console.log('Keyboard requires KeymanWeb 10.0 or later')
```

Am I missing some additional setup?

Reply:
Looks like this has since been resolved?  I don't see a failure here when testing locally.

Reply:
Yeah, when I added @mcdurdin 's changes from 
https://github.com/keymanapp/keyman/pull/5389#discussion_r663562075

it seems to have fixed this unit test